File: New/new.cpp
// #include <iostream>
// #include <cstdio>
// #include <cstdlib>
// #include <cstring>
// #include <cmath>
// #include <algorithm>
// #include <conio.h>
// using namespace std;
// char map[100][100]=
// {
//     "################################",
//     "#OX                            #",
//     "#                              #",
//     "#                              #",
//     "#                              #",
//     "#                              #",
//     "#                  S           #",
//     "#                              #",
//     "#                              #",
//     "#                              #",
//     "################################",
// };
// int sx=1,sy=1;
// int Score=0;

// int score(int x,int y,int d)
// {
//     if(map[x][y]=='X')
//     {
//         if(d==1)
//         {
//             map[x-1][y]='X';
//             sx--;
//         }
//         else if(d==2)
//         {
//             map[x][y-1]='X';
//             sy--;
//         }
//         else if(d==3)
//         {
//             map[x+1][y]='X';
//             sx++;
//         }
//         else if(d==4)
//         {
//             map[x][y+1]='X';
//             sy++;
//         }
//         return 1;
//     }  
//     return 0;
// }
// void print()
// {
//     system("cls");
//     for(int i=0;i<11;i++)
//     {
//         cout<<map[i]<<endl;
//     }
//     printf("---The score you get is %d---\n",Score);

// }
// bool ok(int x,int y)
// {
//     if(map[x][y]=='#')
//     return 0;
//     return 1;
// }
// void input()
// {
    
// }
// int main()
// {
//     while(1)
//     {
//         char c=getch();
//     if(c=='w')
//     {
//         if(!ok(sx-1,sy))
//         continue;
//         if(score(sx-1,sy,1))
//         {
//             Score++;
//         }
//         map[sx][sy]=' ';
//         map[sx-1][sy]='O';
//         sx--;
//         print();
//     }
//     else if(c=='a')
//     {
//         if(!ok(sx,sy-1))
//         continue;
//         if(score(sx,sy-1,2))
//         {
//             Score++;
//         }
//         map[sx][sy]=' ';
//         map[sx][sy-1]='O';
//         sy--;
//         print();
//     }
//     else if(c=='s')
//     {
//         if(!ok(sx+1,sy))
//         continue;
//         if(score(sx+1,sy,3))
//         {
//             Score++;
//         }
//         map[sx][sy]=' ';
//         map[sx+1][sy]='O';
//         sx++;
//         print();
//     }
//     else if(c=='d')
//     {
//         if(!ok(sx,sy+1))
//         continue;
//         if(score(sx,sy+1,4))
//         {
//             Score++;
//         }
//         map[sx][sy]=' ';
//         map[sx][sy+1]='O';
//         sy++;
//         print();
//     }
//     }   
//     return 0;
// }
#include <iostream>
#include <random>
using namespace std;
int a,b,c,d;
void start()
{
    cout<<"----------------"<<endl;
    cout<<"     Welcome    "<<endl;
    cout<<"----------------"<<endl;
    a=rand()%11;
    b=rand()%11;
    c=rand()%11;
    d=rand()%11;
    cout<<a<<" "<<b<<" "<<c<<" "<<d<<endl;
}
int main()
{
    while(1)
    {
        start();

    }
    return 0;
}

File: New/new.py
import random

n1=1
n2=1
n3=1
n4=1


while(1):
    print("--------------------------------")
    print("             Welcome")
    print("--------------------------------")
    print("每次选择一个数字，让它减1，")
    print("谁先把四个数减到0为赢")
    ok=1
    while(ok):
        A=a=random.randint(1,10)
        B=b=random.randint(1,10)
        C=c=random.randint(1,10)
        D=d=random.randint(1,10)
        n1=a%2+b%2+c%2+d%2
        a=int(a/2)
        b=int(b/2)
        c=int(c/2)
        d=int(d/2)
        n2=a%2+b%2+c%2+d%2
        a=int(a/2)
        b=int(b/2)
        c=int(c/2)
        d=int(d/2)
        n3=a%2+b%2+c%2+d%2
        a=int(a/2)
        b=int(b/2)
        c=int(c/2)
        d=int(d/2)
        n4=a%2+b%2+c%2+d%2
        a=int(a/2)
        b=int(b/2)
        c=int(c/2)
        d=int(d/2)
        if(n1%2==0):
            if(n2%2==0):
                if(n3%2==0):
                    if(n4%2==0):
                        ok=0
                        break
    print("Good luck!!!")
    print(A,B,C,D)
    S=A+B+C+D
    step=1;
    
    while(S>0):
        if(step%2==1)
            print("It is your turn!")
            s=input()
            step=step+1
            if(s==1)
                A=A-1
            if(s==2)
                B=B-1
            if(s==3)
                C=C-1
            if(s==4)
                D=D-1
        print('#',step,':',A,B,C,D)
        if(step%2==0)

            print()

    break